snapshotrepository_test.go

```diff
@@ -0,0 +1,177 @@
+package eventsourcing_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/hallgren/eventsourcing"
+	"github.com/hallgren/eventsourcing/eventstore/memory"
+	snap "github.com/hallgren/eventsourcing/snapshotstore/memory"
+)
+
+func TestSaveAndGetSnapshot(t *testing.T) {
+	eventrepo := eventsourcing.NewEventRepository(memory.Create())
+	eventrepo.Register(&Person{})
+
+	snapshotrepo := eventsourcing.NewSnapshotRepository(snap.Create(), eventrepo)
```

Comment:
There is some repetition of code that is setup. Could it be refactored into a setup function that returns all that is needed to drive the test?

Reply:
Yes, the plan is to do something similar to the event store test suite.

Reply:
Is it more clear now?